pandas/tests/reshape/test_concat.py

```diff
@@ -1243,6 +1243,29 @@ def test_concat_dict(self):
         expected = concat([frames[k] for k in keys], keys=keys)
         tm.assert_frame_equal(result, expected)
 
+    def test_concat_mapping(self, non_mapping_dict_subclass):
```

Comment:
can we reuse test_concat_dict using dict and non_mapping_dict_subclass for the constructor using pytest.mark.parametrize?

Reply:
Let me check that.

Reply:
does this work?

```python
    @pytest.mark.parametrize("mapping", ["mapping", "dict"])
    def test_concat_dict(self, mapping, non_mapping_dict_subclass):
        constructor = non_mapping_dict_subclass if mapping == "mapping" else dict
        frames = constructor({
            "foo": DataFrame(np.random.randn(4, 3)),
            "bar": DataFrame(np.random.randn(4, 3)),
            "baz": DataFrame(np.random.randn(4, 3)),
            "qux": DataFrame(np.random.randn(4, 3)),
        })
```

Reply:
Neat workaround! Let me update the PR.